### PR TITLE
Add etcd discovery flags

### LINKED ISSUE #1956: New etcd discovery options: --etcd-discovery and --etcd-discovery-srv
https://github.com/sensu/sensu-go/pull/3417 adds etcd and DNS SRV discovery options for the embedded etcd server we use in `sensu-backend`.

## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [X] New content (guide wanted)
- [ ] Update (Add missing or refresh existing content)
- [ ] Enhancement (add new site functionality)

## Expected Behavior

The 2 new options and what they do are described in our documentation. See PR above for links that should give a bit of context.

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Add `etcd-discovery` and `etcd-discovery-srv` flags to backend reference doc.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/1956

## Review Instructions
- [x] Not sure if the URL used in the examples are correct.
- [ ] Not sure if other documentation changes are needed. For example, should this be added to the federation API doc for 5.16?

### CODE REVIEW

File: content/sensu-go/5.16/reference/backend.md [hunk 1]
```diff
@@ -810,7 +842,7 @@ eventd-workers: 100{{< /highlight >}}
 
 | keepalived-buffer-size |      |
 -----------------------|------
-description            | Number of incoming keepalives that can be buffered before being processed by a keepalived worker. _WARNING: Modify with caution. Increasing this value may result in higher memory usage._
+description            | Number of incoming keepalives that can be buffered before beinghttps://etcd.io/docs/v3.3.12/op-guide/clustering/#dns-discovery processed by a keepalived worker. _WARNING: Modify with caution. Increasing this value may result in higher memory usage._
```
Comment (ccressent) [inline content/sensu-go/5.16/reference/backend.md]: I'm assuming this hunk should be discarded?

File: content/sensu-go/5.16/reference/backend.md [hunk 2]
```diff
@@ -605,6 +609,34 @@ sensu-backend start --etcd-client-cert-auth
...
+{{< /highlight >}}
+
+
+| etcd-discovery-srv    |      |
+------------------------|------
+description             | Exposes [etcd's embedded auto-discovery features][17]. Attempts to use a [DNS SRV][19] record to get the cluster configuration.
+type                    | String
+default                 | ""
+example                 | {{< highlight shell >}}# Command line example
+sensu-backend start --etcd-discovery-srv https://10.0.0.1:2380
```
Comment (ccressent) [inline content/sensu-go/5.16/reference/backend.md]: In the case of DNS SRV, the argument is simply a domain name, such as `example.org` or `test.com`. Etcd then uses the [SRV record](https://en.wikipedia.org/wiki/SRV_record) of that domain name to discover the cluster configuration.

File: content/sensu-go/5.16/reference/backend.md [hunk 3]
```diff
@@ -605,6 +609,34 @@ sensu-backend start --etcd-client-cert-auth
...
+------------------------|------
+description             | Exposes [etcd's embedded auto-discovery features][17]. Attempts to use [etcd discovery][18] to get the cluster configuration.
+type                    | String
+default                 | ""
+example                 | {{< highlight shell >}}# Command line example
+sensu-backend start --etcd-discovery https://10.0.0.1:2380
+
+# /etc/sensu/backend.yml example
+etcd-discovery:
+  - https://10.0.0.1:2380
```
Comment (ccressent) [inline content/sensu-go/5.16/reference/backend.md]: For this discovery method, I think you should reuse one of the example URLs used in the Etcd docs, e.g.: `https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de`.

**Review by ccressent — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.16/reference/backend.md (hunk 1 above)
Comment (hillaryfraley) [inline content/sensu-go/5.16/reference/backend.md]:
```suggestion
description            | Number of incoming keepalives that can be buffered before being processed by a keepalived worker. _WARNING: Modify with caution. Increasing this value may result in higher memory usage._
```

Comment (hillaryfraley) [reply content/sensu-go/5.16/reference/backend.md]: Yes, I must have pasted it there without realizing. Thank you!

File: content/sensu-go/5.16/reference/backend.md (hunk 3 above)
Comment (hillaryfraley) [inline content/sensu-go/5.16/reference/backend.md]:
```suggestion
  - https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de
```

File: content/sensu-go/5.16/reference/backend.md [hunk 4]
```diff
@@ -605,6 +609,34 @@ sensu-backend start --etcd-client-cert-auth
 etcd-client-cert-auth: true{{< /highlight >}}
 
 
+| etcd-discovery        |      |
+------------------------|------
+description             | Exposes [etcd's embedded auto-discovery features][17]. Attempts to use [etcd discovery][18] to get the cluster configuration.
+type                    | String
+default                 | ""
+example                 | {{< highlight shell >}}# Command line example
+sensu-backend start --etcd-discovery https://10.0.0.1:2380
```
Comment (hillaryfraley) [inline content/sensu-go/5.16/reference/backend.md]:
```suggestion
sensu-backend start --etcd-discovery https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de
```

File: content/sensu-go/5.16/reference/backend.md (hunk 2 above)
Comment (hillaryfraley) [inline content/sensu-go/5.16/reference/backend.md]:
```suggestion
sensu-backend start --etcd-discovery-srv example.org
```

File: content/sensu-go/5.16/reference/backend.md [hunk 5]
```diff
@@ -605,6 +609,34 @@ sensu-backend start --etcd-client-cert-auth
...
+------------------------|------
+description             | Exposes [etcd's embedded auto-discovery features][17]. Attempts to use a [DNS SRV][19] record to get the cluster configuration.
+type                    | String
+default                 | ""
+example                 | {{< highlight shell >}}# Command line example
+sensu-backend start --etcd-discovery-srv https://10.0.0.1:2380
+
+# /etc/sensu/backend.yml example
+etcd-discovery-srv:
+  - https://10.0.0.1:2380
```
Comment (hillaryfraley) [inline content/sensu-go/5.16/reference/backend.md]:
```suggestion
  - example.org
```

**Review by ccressent — APPROVED** [review]
Looks good to me!

As for your question about adding content in an existing or new guide: I think the flag/config reference is a good start and probably enough for now. A guide about using this etcd feature might be out of scope given that it's a pure etcd thing?

@cwjohnston may have another opinion or idea though.

**Comment by hillaryfraley** [comment]
@cwjohnston  -- conceivable that we would update the clustering docs, but that is not in scope for this release. OK to merge this PR.

**Pull request merged**